elytron/src/main/java/org/wildfly/extension/elytron/ProviderDefinitions.java

```diff
@@ -236,11 +239,21 @@ protected void resolveRuntime(ModelNode model, OperationContext context) throws
                                             Constructor<? extends Provider> constructor = doPrivileged(
                                                     (PrivilegedExceptionAction<Constructor<? extends Provider>>) () -> providerClazz
                                                             .getConstructor(InputStream.class));
-                                            loadedProviders.add(constructor.newInstance(new Object[] { configSupplier.get() }));
-                                        } catch (NoSuchMethodException ignored) {
-                                            Provider provider = providerClazz.newInstance();
-                                            loadedProviders.add(provider);
-                                            deferred.add(provider::load);
+                                            loadedProviders.add(constructor.newInstance(new Object[]{configSupplier.get()}));
+                                        } catch (NoSuchMethodException constructorDoesNotExist) {
+                                            File tempFile = null;
+                                            try {
+                                                Method configureMethod = doPrivileged((PrivilegedExceptionAction<Method>) () -> providerClazz.getMethod("configure", String.class));
+                                                Provider provider = providerClazz.newInstance();
+                                                tempFile = inputStreamToFile(configSupplier);
+                                                loadedProviders.add((Provider) configureMethod.invoke(provider, tempFile.getAbsolutePath()));
+                                            } catch (NoSuchMethodException configureMethodDoesNotExist) {
+                                                Provider provider = providerClazz.newInstance();
+                                                loadedProviders.add(provider);
+                                                deferred.add(provider::load);
+                                            } finally {
+                                                tempFile.delete();
```

Comment:
Thanks @Skyllarr! I think just a null check is needed here now.

Reply:
@fjuma Fixed. Thank you